test/utilities_tests.jl

```diff
@@ -19,4 +21,51 @@ for FT in (Float32, Float64)
         @test parent(field1) == parent(field2)
         @test axes(field2) == space2
     end
+
+    @testset "test comms_cntx cpu" begin
+        parsed_args = Dict("device" => "auto")
+
+        this_device = ClimaComms.device()
+        this_ctx = ClimaComms.context(this_device)
+
+        @test typeof(Utilities.get_comms_context(parsed_args)) == typeof(this_ctx)
+
+        parsed_args["device"] = "CUDADevice"
+        @test typeof(Utilities.get_comms_context(parsed_args)) == typeof(ClimaComms.context(ClimaComms.CUDADevice()))
+
+        parsed_args["device"] = "CPUMultiThreaded"
+        @test typeof(Utilities.get_comms_context(parsed_args)) ==
+              typeof(ClimaComms.context(ClimaComms.CPUMultiThreaded()))
+
+        parsed_args["device"] = ""
+        @test typeof(Utilities.get_comms_context(parsed_args)) ==
+              typeof(ClimaComms.context(ClimaComms.CPUSingleThreaded()))
+
+        ClimaComms.nprocs(Utilities.get_comms_context(parsed_args))
+
+        Utilities.comms_context_info(nothing, nothing)
```

Comment:
It would be better to do have all of the test coverage in one testset, instead of adding another testset just for two additional calls. We should also add a comment explaining that since Github Actions only exercises the SingleThreaded case, we have to add these extra function calls for code coverage (even though the code is meaningfully tested on buildkite).

I still don't understand why we need the extra `ClimaComms.nprocs` call - won't that be called in `get_comms_context` every time? It's also a function that isn't in our repo so we shouldn't have to test it

Reply:
Okay, I can combine them into one test set and add a comment. 
Without the ClimaComms.nprocs call the function was missing the else statement: [https://github.com/CliMA/ClimaCoupler.jl/blob/debc5defe80fa8a956ddabc29b9496e6d3780c53/src/Utilities.jl#L79-L81](url)

Reply:
I thought about it some more and I don't think it's good practice to write tests just to increase our code coverage, when the tests themselves aren't meaningful/necessary. @LenkaNovak is it okay with you to remove the tests under the "Additional test calls ..." comment?

Reply:
I would argue that this is still testing that if `get_comms_context` is passed a particular `parsed_args["device"]` it returns the correct context, so I'd be inclined to keep it. 

In fact, I wonder if we should implement a warning in `get_comms_context` if `device = get_device(parsed_args)` is not `ClimaComms.device()`.

Reply:
That's a good point. Maybe we could just remove the last two function calls then (`nprocs` and `comms_context_info(nothing, nothing)`) since those are really just for code coverage.

I feel ambivalent about the warning - I think either way would be fine